cookbook/knowledge/filters/README.md

```diff
@@ -0,0 +1,211 @@
+
+```python
+from agno.knowledge.knowledge import Knowledge
+from agno.vectordb.lancedb import LanceDb
+from agno.db.postgres.postgres import PostgresDb
+
+# Initialize vector and contents databases
+vector_db = LanceDb(
+    table_name="recipes",
+    uri="tmp/lancedb",
+)
+
+knowledge = Knowledge(
+    vector_db=vector_db,
+    max_results=5,
+    contents_db=PostgresDb(
+        table_name="content_store",
+        host="localhost",
+        port=5432,
+        username="ai",
+        password="ai",
+        database="ai",
+    ),
+)
+
+# Load content with specific criteria
+knowledge.add_contents(
+    documents_to_add,
+    metadata_key="file_name"
+)
+```
+
+### 3. Async Filtering (`async_filtering.py`)
+
+**When to use**: High-performance applications requiring concurrent filtering operations.
+
+**How it works**: Asynchronous processing enables parallel filtering across multiple data sources.
+
+```python
+import asyncio
+from agno.knowledge.knowledge import Knowledge
+from agno.vectordb.lancedb import LanceDb
+
+async def async_filtering_example():
+    vector_db = LanceDb(
+        table_name="async_recipes",
+        uri="tmp/lancedb",
+    )
+    
+    knowledge = Knowledge(
+        name="Async Knowledge Base",
+        vector_db=vector_db,
+    )
+    
+    # Async content loading
+    await knowledge.add_content(
+        path="data/documents",
+    )
+    
+    # Async search with filters
+    results = await knowledge.async_search(
+        query="your query",
+        metadata_filters={"category": "recipes"}
+    )
+    
+    return results
+
+# Run async example
+results = asyncio.run(async_filtering_example())
+```
+
+### 4. Agentic Filtering (`agentic_filtering.py`)
+
+**When to use**: Complex filtering logic requiring AI understanding and decision-making.
+
+**How it works**: Uses AI agents to intelligently filter content based on semantic understanding and context.
+
+```python
+from agno.agent import Agent
+from agno.knowledge.knowledge import Knowledge
+from agno.vectordb.lancedb import LanceDb
+
+# Create knowledge base
+vector_db = LanceDb(table_name="agentic_recipes", uri="tmp/lancedb")
+knowledge = Knowledge(name="Agentic Filtered KB", vector_db=vector_db)
+
+# Create filtering agent
+filter_agent = Agent(
+    name="Content Filter Agent",
+    role="Intelligent content filtering specialist", 
+    instructions=[
+        "Analyze documents for relevance, quality, and appropriateness",
+        "Apply business rules and contextual understanding",
+        "Provide filtering rationale for transparency"
+    ],
+    knowledge=knowledge,
+    search_knowledge=True,
+)
+
+# Load content and let agent filter intelligently
+knowledge.add_contents(documents_to_add)
+
+# Agent can now intelligently filter and retrieve content
+response = filter_agent.run("Find high-quality technical documentation")
+```
+
+### 5. Invalid Keys Handling (`filtering_with_invalid_keys.py`)
```

Comment:
This is a very atypical pattern. We want users to only really use filtering inside the agent. But I guess this is fine to have